applications/FluidDynamicsApplication/custom_processes/mass_conservation_check_process.cpp

```diff
@@ -200,8 +200,18 @@ void MassConservationCheckProcess::ComputeVolumesAndInterface( double& positiveV
             Vector Distance( rGeom.PointsNumber(), 0.0 );
             for (unsigned int i = 0; i < rGeom.PointsNumber(); i++){
                 // Control mechanism to avoid 0.0 ( is necessary because "distance_modification" possibly not yet executed )
-                if ( rGeom[i].FastGetSolutionStepValue(DISTANCE) == 0.0 ){
-                    it_elem->GetGeometry().GetPoint(i).FastGetSolutionStepValue(DISTANCE) = 1.0e-7;
+                const double dist = rGeom[i].FastGetSolutionStepValue(DISTANCE);
+                if ( std::abs(dist) < 1.0e-15 ){
```

Comment:
I think this tolerance is too close to the machine precision limit. I'd set something around 1.0e-12.

Reply:
The point is that originally it was exactly zero "0.0" (let's say it is a check for zero). I think it would work fine since the threshold would become 1.0e6*NODAL_H.

Reply:
If you need to check that the distance is non-zero I'd check with `std::numeric_limits<double>::min()`. If you want to check that the distance value is extremely small, I wouldn't use 1.0e-15 as it's really close to the numeric limits (in this case I assume that what we want to do is the second option so it's safer to use a larger threshold).